pkg/apis/stork/v1alpha1/applicationrestore.go

```diff
@@ -0,0 +1,126 @@
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	// ApplicationRestoreResourceName is name for "applicationrestore" resource
+	ApplicationRestoreResourceName = "applicationrestore"
+	// ApplicationRestoreResourcePlural is plural for "applicationrestore" resource
+	ApplicationRestoreResourcePlural = "applicationrestores"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+// ApplicationRestore represents applicationrestore object
+type ApplicationRestore struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Spec              ApplicationRestoreSpec   `json:"spec"`
+	Status            ApplicationRestoreStatus `json:"status"`
+}
+
+// ApplicationRestoreSpec is the spec used to restore applications
+type ApplicationRestoreSpec struct {
+	BackupName       string                              `json:"backupName"`
+	BackupLocation   string                              `json:"backupLocation"`
+	NamespaceMapping map[string]string                   `json:"namespaceMapping"`
+	Selectors        map[string]string                   `json:"selectors"`
+	EncryptionKey    *corev1.EnvVarSource                `json:"encryptionKey"`
+	ReplacePolicy    ApplicationRestoreReplacePolicyType `json:"replacePolicy"`
+}
+
+// ApplicationRestoreReplacePolicyType is the replace policy for the application restore
+// in case there are conflicting resources already present on the cluster
+type ApplicationRestoreReplacePolicyType string
+
+const (
+	// ApplicationRestoreReplacePolicyDelete is to specify that the restore
+	// should delete existing resources that conflict with resources being
+	// restored
+	ApplicationRestoreReplacePolicyDelete ApplicationRestoreReplacePolicyType = "Delete"
+	// ApplicationRestoreReplacePolicyRetain is to specify that the restore
+	// should retain existing resources that conflict with resources being
+	// restored
+	ApplicationRestoreReplacePolicyRetain ApplicationRestoreReplacePolicyType = "Retain"
+	// ApplicationRestoreReplacePolicyFail is to specify that the restore
+	// should fail if there are existing resources that conflict with resources being
+	// restored
+	ApplicationRestoreReplacePolicyFail ApplicationRestoreReplacePolicyType = "Fail"
```

Comment:
As discussed offline, may be we don't need to support Fail policy. Will be simpler to implement and no need to cleanup on partial restores.

Reply:
Removed for now, I wasn't using it in the controller anyways when I implemented it.